src/auth.rs

```diff
@@ -13,18 +13,20 @@ pub struct Auth {
 }
 
 impl Auth {
-    pub fn graph(client: Arc<Client>) -> Self {
+    pub fn new(client: Option<Arc<Client>>, resource: &'static str) -> Self {
```

Comment:
An optional client is really confusing. Should we make fn new private and allow to take non-optional client?
```suggestion
    fn new(client: Arc<Client>, resource: &'static str) -> Self {
```

Reply:
To be honest, this was entirely to prevent having to also specify `reqwest` as a dependency: not that it is bad, but because we would only be using `reqwest::Client` for this purpose (and for nothing else). However, this is not a major issue.

I think it is best to remove the `Option`, but would it be possible for the method to remain public? This way, others could use `Auth` to access audiences besides `https://graph.microsoft.com` and `https://management.azure.com`.

Reply:
Ok. Makes sense